### ISSUE TITLE
How to preprocess for TIMIT RNN?

### ISSUE THREAD

**Issue by Originofamonia** [mravanelli/pytorch-kaldi#176]
Hello!
I could run the TIMIT_MLP_mfcc_basic. 
So I was wondering what would be the appropriate preprocess commands to run TIMIT_RNN_mfcc? 
It seems to need 3 kinds of features (mfcc, FBANK, fMLLR). So how could I get all 3 kinds of features and why we need all 3 kinds of features here instead of only one feature in the TIMIT_MLP_mfcc_basic?
Thanks in advance!
Edward

**Comment by mravanelli** [comment]
If you take a look into the [model] section (out_dnn1=compute(RNN_layers,mfcc),
we finally use only MFCCs.
In the data section we declare all the features such that it is easy to
switch from one to the other, but you can cancel the declaration of the
other features without problems.
Also, I don't suggest you to use RNN. There are better models available,
such as the LiGRU one.

Best,

Mirco

**Comment by Originofamonia** [comment]
Thanks for your reply!
How about the labels? I saw both RNN and LiGRU use label mono, while MLP basic uses label cd. Are they the same kind of labels?
Another question might be not related to pytorch-kaldi is I found most papers are doing PER instead of WER on the TIMIT dataset. Is there a particular reason for that? Or just because TIMIT provided the phoneme level labels?

Thanks!
Edward

**Comment by TParcollet** [comment]
Hi! Labels mono are used to perform the so-called "monoplane regularization" that helps the training. RNN and LiGRU also use label_cd (cd = Context Dependent), that are the HMM states that we want to produce for decoding. TIMIT is in PER because labels are Phonemes. in Pytorch-Kaldi results are reported in PER, even if it's written WER (due to the Kaldi decoding script).

**Comment by Originofamonia** [comment]
When I tried to debug the TIMIT RNN, I found the input data has dimension [26, 8, 39]. I know the batch size is 8 and each mfcc feature is a [3, 13] matrix making 39. So what does the 26 mean?
Thanks!

**Comment by mravanelli** [comment]
The standard for RNNs is n-time-steps, batch-size, feature-dim
